packages/cli/src/util/update-notifier/update-worker.js

```diff
@@ -20,7 +20,14 @@ process.on('unhandledRejection', err => {
   process.exit(1);
 });
 
-process.on('message', async msg => {
+// this timer will prevent this worker process from running longer than 10s
+const timer = setTimeout(() => {
+  console.error('Update worker timed out, exiting');
```

Comment:
**suggestion (soft-blocking):** move to debug log

In this or a follow-up PR: Can wove this to a debug log? Because the worker inherits stdio from the parent, if the `vc` command takes >= 10 seconds and the work does time out, the user will see this message. That would be confusing and not actionable for them.

Reply:
I thought about that, but I don't see why it's necessary. The worker will have disconnected and the parent will have exited long before that timeout fires.

Besides, we'd have to create a new `output` instance and then we'd need to figure out if we should pass `--debug` through from the parent or parse it again in the worker.

Maybe we just switch `stdio` to `['ignore', 'ignore', 'ignore', 'ipc']` or just remove the log message?

Reply:
I agree that's it's more likely that the user will never see this, but a bad connection could still show this log.

We could pass a debug flag in (probably already parsed into a boolean) and only log if that's true. I wouldn't create a new `output` for this.

But if we don't want to debug log this, I'd rather remove the log entirely.

Reply:
Bad connection? The IPC channel? I think once it's disconnected, it's disconnected. Maybe it's possible the stdout/stderr file descriptors are still valid even after the child process has disconnected.

Yeah, I'll just rip out the log instead of passing flags and such. Probably one of the best solutions is to log all debug to a file in the update cache directory, but I don't think that's necessary.

Reply:
I mean a bad internet connection. It could cause the CLI command (like `vc deploy`) to take a while as well as the call to look up the latest CLI version. If both of those take 10+ seconds, this `console.log` would fire.

Reply:
Ah, now I'm tracking.

FWIW, just as I suspected, even if the IPC channel is closed and the parent exits, the child still has a valid stdout/stderr file descriptors. Setting `stdio` to `"ignore"` will not pass the stdio file descriptors to the child and thus any console logs from the child will not be displayed.